File: Completed/JavaScript/169.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
  const map = {}
  
  for (let num of nums) {
      if (map[num]) {
          map[num] += 1
      } else {
          map[num] = 1
      }
      
      if (map[num] > nums.length /2) return num
  }
};

File: Completed/Go/169.go

//O(N) space and time
func majorityElement(nums []int) int {
	m := map[int]int{}
	for _, num := range nums {
		m[num] += 1

		if m[num] > len(nums)/2 {
			return num
		}
	}

	return -1
}

// O(N) time

import "sort"

func majorityElement(nums []int) int {
    sort.Ints(nums)
    return nums[len(nums)/2]
}


